### PR TITLE
feat: Added src-divider

### PR DESCRIPTION
**Pull request by oliverlloyd**
# `Divider`

Inserts a line on the page acting as a section break dividing content

## Install

```sh
$ yarn add @guardian/src-divider
```

## Use

```tsx
import { Divider } from '@guardian/src-divider';

const Content = () => (
    <>
        <p>Some text</p>
        <Divider />
        <p>Some text</p>
        <Divider
            size="small"
            spaceAbove="loose"
            displayText="I am centered inside the divider"
        />
        <p>Some text</p>
    </>
);
```

### `size`

**`'partial' | 'full'`**

`full` means the divider will span the full width of the content
`partual` left aligns the divider and limits its length

### `spaceAbove`

**`'tight' | 'loose'`**

How much space is given above the divider

### `displayText`

**`string`**

This text is centered inside the dividing line

### CODE REVIEW

**Comment by oliverlloyd** [comment]
Relates to https://github.com/guardian/source/discussions/814

File: src/core/components/divider/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+## Divider Props
+
+### `size`
+
+**`'partial' | 'full'` **
+
+`full` means the divider will span the full width of the content
+`partual` left aligns the divider and limits its length
```
Comment (SiAdcock) [inline src/core/components/divider/README.md:38]:
```suggestion
`partial` left aligns the divider and limits its length
```

File: src/core/components/divider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { space } from '@guardian/src-foundations';
+import { textSans } from '@guardian/src-foundations/typography';
+
+type Props = {
+	size?: 'full' | 'partial';
+	spaceAbove?: 'tight' | 'loose';
+	displayText?: string;
+};
+
+const fullStypes = css`
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:13]:
```suggestion
const fullStyles = css`
```

File: src/core/components/divider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	return (
+		<hr
+			css={[
+				css`
+					height: 1px;
+					border: 0;
+					margin-bottom: 3px;
+					background-color: ${border.secondary};
+				`,
+				size === 'partial' ? partialStyles : fullStypes,
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:100]:
```suggestion
				size === 'partial' ? partialStyles : fullStyles,
```

File: src/core/components/divider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+const fullStyles = css`
+	margin-left: -10px;
+	width: calc(100% + 20px);
+`;
+const partialStyles = css`
+	margin-left: -10px;
+	width: 160px;
+`;
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:20]: Please can we move the styles to a separate `styles.ts` file in the `divider` directory?

File: src/core/components/divider/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+const decideSpace = (
+	spaceAbove: 'tight' | 'loose',
+	displayText?: string,
+): SerializedStyles => {
+	switch (spaceAbove) {
+		case 'tight':
+			return displayText
+				? css`
+						margin-top: -3px;
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:30]:
```suggestion
						margin-top: -space[1]px;
```

File: src/core/components/divider/index.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+				  `
+				: css`
+						margin-top: ${space[6]}px;
+				  `;
+		case 'loose':
+			return displayText
+				? css`
+						margin-top: 37px;
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:38]:
```suggestion
						margin-top: ${space[9]}px;
```

File: src/core/components/divider/index.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		);
+	}
+
+	return (
+		<hr
+			css={[
+				css`
+					height: 1px;
+					border: 0;
+					margin-bottom: 3px;
```
Comment (SiAdcock) [inline src/core/components/divider/index.tsx:97]:
```suggestion
					margin-bottom: ${space[1]}px;
```

**Comment by oliverlloyd** [comment]
@SiAdcock where did we end up on the layout / not layout discussion? Are you okay keeping this divider as it's own thing?

**Comment by SiAdcock** [comment]
@oliverlloyd what do you think about adding this to the Development Kitchen (#963)?

**Comment by oliverlloyd** [comment]
> what do you think about adding this to the Development Kitchen

👍 I'll cook something up

**Comment by sndrs** [comment]
> 👍 I'll cook something up

👨‍🍳👌

**Pull request merged**